### PR TITLE
Added sort and thenSort directives to FunctionCompiler

### PR DESCRIPTION
**Pull request by drakeet**
Although can use `morph (new SortList<T>())` to achieve the same
function, but I think sort is a very common function, should not let
each person alone to achieve a `SortList` Function. So I added the `sort` and `thenSort`
directives to make it more convenient.

### CODE REVIEW

**Comment by drakeet** [comment]
I signed it!

File: agera/src/main/java/com/google/android/agera/FunctionCompilerStates.java [hunk 1]
```diff
@@ -113,6 +114,14 @@
     FList<TPrev, TPrevList, TFrom> limit(int limit);
 
     /**
+     * Adds a {@link Comparator} to the behavior chain to sort the items.
+     *
+     * @param comparator the comparator to sort the items
+     */
+    @NonNull
+    FList<TPrev, TPrevList, TFrom> sort(Comparator<TPrev> comparator);
```
Comment (ernstsson) [inline agera/src/main/java/com/google/android/agera/FunctionCompilerStates.java]: `@NonNull` on the param

File: agera/src/main/java/com/google/android/agera/FunctionCompilerStates.java [hunk 2]
```diff
@@ -127,5 +136,13 @@
...
     @NonNull
     Function<TFrom, TPrevList> thenLimit(int limit);
+
+    /**
+     * Adds a {@link Comparator} to the behavior chain to sort the items.
+     *
+     * @param comparator the comparator to sort the items
+     */
+    @NonNull
+    Function<TFrom, TPrevList> thenSort(Comparator<TPrev> comparator);
```
Comment (ernstsson) [inline agera/src/main/java/com/google/android/agera/FunctionCompilerStates.java]: `@NonNull` on the param

File: agera/src/main/java/com/google/android/agera/FunctionCompiler.java [hunk 3]
```diff
@@ -209,4 +225,20 @@ public Object apply(@NonNull final Object input) {
       return result;
     }
   }
+
+  private static final class SortFunction<T> implements Function<List<T>, List<T>> {
+    @NonNull
+    private final Comparator comparator;
+
+    public SortFunction(Comparator comparator) {
```
Comment (ernstsson) [inline agera/src/main/java/com/google/android/agera/FunctionCompiler.java]: `@NonNull final` and `checkNotNull(comparator)` in the assignment

File: agera/src/main/java/com/google/android/agera/FunctionCompiler.java [hunk 4]
```diff
@@ -94,6 +96,13 @@ private Function createFunction() {
 
   @NonNull
   @Override
+  public FunctionCompilerStates.FList sort(Comparator comparator) {
```
Comment (ernstsson) [inline agera/src/main/java/com/google/android/agera/FunctionCompiler.java]: `@NonNull final`

**Comment by ernstsson** [comment]
Looks pretty cool, just a few minor comments!

**Comment by drakeet** [comment]
Thank you. I've fixed them.

**Pull request merged**